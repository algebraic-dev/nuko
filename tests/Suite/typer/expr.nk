type List a =
  | Cons a (List a)
  | Nil

type Either a b c = | Nel

let add (a: ffff) (b: List b) (c: forall c. List c -> Either b d ffff) (d: List d): Int = 3

let adf (c: forall c. List c -> List b): Int = 3

let ade (a: ffff) (b: List b) (c: Int -> List d) (d: List b): Int = 3

let at (a: Int): Int = 
  let a = 3
  add a a