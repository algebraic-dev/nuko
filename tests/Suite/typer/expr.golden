âœ— This
"CannotUnify "Prelude.Int" "(Main.List ^b.0)""
